test/integration/save_world.cc

```diff
@@ -302,6 +304,130 @@ TEST_F(SdfGeneratorFixture, WorldWithNestedModel)
+  EXPECT_NE(kNullEntity, this->server->EntityByName("L1"));
+  EXPECT_NE(kNullEntity, this->server->EntityByName("C1"));
+  EXPECT_NE(kNullEntity, this->server->EntityByName("V1"));
+  EXPECT_NE(kNullEntity, this->server->EntityByName("include_nested_new_name"));
+  EXPECT_NE(kNullEntity, this->server->EntityByName("link_00"));
+  EXPECT_NE(kNullEntity, this->server->EntityByName("link_01"));
+  EXPECT_NE(kNullEntity, this->server->EntityByName("sphere"));
+  EXPECT_NE(kNullEntity, this->server->EntityByName("V"));
+  EXPECT_NE(kNullEntity, this->server->EntityByName("C"));
+
+  const std::string worldGenSdfRes =
+      this->RequestGeneratedSdf("model_nested_include_world");
+
+  // check that model w/ nested includes are not expanded
+  tinyxml2::XMLDocument genSdfDoc;
+  genSdfDoc.Parse(worldGenSdfRes.c_str());
+  ASSERT_NE(nullptr, genSdfDoc.RootElement());
+  auto genWorld = genSdfDoc.RootElement()->FirstChildElement("world");
+  ASSERT_NE(nullptr, genWorld);
+
+  auto model = genWorld->FirstChildElement("model");  // ground_plane
+  ASSERT_NE(nullptr, model);
+  model = model->NextSiblingElement("model");  // M1
+  ASSERT_NE(nullptr, model);
+
+  // M1's child include
+  auto include = model->FirstChildElement("include");
+  ASSERT_NE(nullptr, include);
+
+  auto uri = include->FirstChildElement("uri");
+  ASSERT_NE(nullptr, uri);
+  ASSERT_NE(nullptr, uri->GetText());
+  EXPECT_EQ("include_nested", std::string(uri->GetText()));
+
+  auto name = include->FirstChildElement("name");
+  ASSERT_NE(nullptr, name);
+  ASSERT_NE(nullptr, name->GetText());
+  EXPECT_EQ("include_nested_new_name", std::string(name->GetText()));
+
+  auto pose = include->FirstChildElement("pose");
+  ASSERT_NE(nullptr, pose);
+  ASSERT_NE(nullptr, pose->GetText());
+
+  std::stringstream ss(pose->GetText());
+  ignition::math::Pose3d p;
+  ss >> p;
+  EXPECT_EQ(ignition::math::Pose3d(1, 2, 3, 0, 0, 0), p);
+
+  // M2
+  model = model->FirstChildElement("model");
+  ASSERT_NE(nullptr, model);
+
+  // M2's child include
+  include = model->FirstChildElement("include");
+  ASSERT_NE(nullptr, include);
+
+  uri = include->FirstChildElement("uri");
+  ASSERT_NE(nullptr, uri);
+  ASSERT_NE(nullptr, uri->GetText());
+  EXPECT_EQ("sphere", std::string(uri->GetText()));
+
+  name = include->FirstChildElement("name");
+  EXPECT_EQ(nullptr, name);
+
+  pose = include->FirstChildElement("pose");
+  ASSERT_NE(nullptr, pose);
+  ASSERT_NE(nullptr, pose->GetText());
+
+  ss = std::stringstream(pose->GetText());
+  ss >> p;
+  EXPECT_EQ(ignition::math::Pose3d(0, 2, 2, 0, 0, 0), p);
+
+  // M3
+  model = model->FirstChildElement("model");
+  ASSERT_NE(nullptr, model);
+
+  // M3's child include
+  include = model->FirstChildElement("include");
+  ASSERT_NE(nullptr, include);
+
+  uri = include->FirstChildElement("uri");
+  ASSERT_NE(nullptr, uri);
+  ASSERT_NE(nullptr, uri->GetText());
+  EXPECT_EQ(
+    "https://fuel.ignitionrobotics.org/1.0/OpenRobotics/models/Coke Can",
```

Comment:
Saving the Fuel version number doesn't seem to work for me. Can you add a test to verify?

Reply:
I modified the previous test, what do you think? 57d583390c6c9d3f39f520f0570a9637f5060fcd